src/jshint.js

```diff
@@ -4732,14 +4732,15 @@ var JSHINT = (function() {
       name = tkn.value;
     }
 
-    if (props[name] && _.has(props, name)) {
+    if ((props[name] && _.has(props, name)) || (name === "__proto__" &&
+        Object.getPrototypeOf(props) !== Object.prototype)) { // __proto__ has been modified
```

Comment:
Since this is annex B stuff, do you want to make this specific to the `browser` environment?